include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -679,7 +678,127 @@ template <typename Lyt, typename TT>
 
     detail::is_operational_impl<Lyt, TT> p{lyt, spec, params};
 
-    return p.determine_operational_input_patterns();
+    std::set<uint64_t> input_patterns{};
+
+    // all possible input patterns
+    for (auto i = 0u; i < spec.front().num_bits(); ++i)
+    {
+        input_patterns.insert(i);
+    }
+
+    const auto non_op_patterns_and_op_status = p.determine_non_operational_input_patterns_and_op_status_due_kinks();
+
+    for (const auto& pattern : non_op_patterns_and_op_status)
+    {
+        input_patterns.erase(pattern.first);
+    }
+
+    return input_patterns;
+}
+
+/**
+ * This function determines the input combinations for which the SiDB layout are non-operational due to kinks.
+ *
+ * @tparam Lyt Type of the cell-level layout.
+ * @tparam TT Type of the truth table.
```

Comment:
Be consistent with your docstrings:

`@tparam TT Type of the truth table.`

vs.

`@tparam TT The type of the truth table specifying the layout behavior.`
 
 etc.